lib/liquid/template.rb

```diff
@@ -64,11 +62,11 @@ class << self
       attr_writer :taint_mode
 
       def file_system
-        @@file_system
+        Thread.current[:liquid_file_system] ||= BlankFileSystem.new
```

Comment:
I think it never returns nil, if `Thread.current[:liquid_file_system]` is nil it will assign `BlankFileSystem.new` and return it, same thing for every separate thread.